src/types/index.ts

```diff
@@ -0,0 +1,56 @@
+export type RowCharacter = {
```

Comment:
using a name "index" for this file is a bit too vague. You are not using an index export pattern and this file contains types for API response shape so I'd say it makes more sense to be under `/api`. Be careful of using `index-` for file name as it easily contaminates your project coz they are all the same name so the more you name files `index-` the less searchability your project has.